crates/ruff_cli/src/commands/run.rs

```diff
@@ -80,18 +80,21 @@ pub(crate) fn run(
 
     // Load the caches.
     let caches = bool::from(cache).then(|| {
-        package_roots
-            .values()
-            .flatten()
-            .dedup()
-            .map(|package_root| {
-                let settings = resolver.resolve_all(package_root, pyproject_config);
+        let cache_roots = package_roots
+            .iter()
+            .map(|(package, package_root)| package_root.unwrap_or(package))
+            .unique()
+            .collect::<Vec<_>>();
+        cache_roots
+            .par_iter()
```

Comment:
Why are we collecting only to iterate over it again?

Reply:
None of those structs implement `.par_iter()`.